### PR TITLE
[WIP] Add AppVeyor config to run tests on win32

### PR DESCRIPTION
**Pull request by tmm1**
wip, don't merge yet

### CODE REVIEW

**Comment by Varriount** [comment]
Um, isn't this rather redundant considering that we have buildbots already? Granted, they don't have the ability to send checks to pull requests, but that's merely due to lack of time.

**Comment by tmm1** [comment]
Like travis, appveyor will run tests on any fork that creates a PR. Like this one for instance, is in my fork of nim-lang.

Their PR integration is nice, plus they have a large build farm which means tests will complete faster and that's one less thing for us to maintain and fix when it breaks.

Anyway, I'm just experimenting for now (so far it's much simpler to iterate on the appveyor.yml than it was to write buildbot specs, and I am able to test my changes directly in this PR instead of needing access to deploy to buildbot separately).

**Comment by Varriount** [comment]
Boudoir can run pull requests as well, I just haven't set out to do so...

Ah well. Does Appveyor support IRC notifications?

**Comment by dom96** [comment]
This is brilliant! I think we're better off moving away from hosting our own build farm and making use of travis, circle, or appveyor.

Does appveyor give us a way to download the resulting binaries/docs? I would love to set that up. It was my original intent with NimBuild as I don't think Travis supports that, Circle might nowadays(?).

**Comment by dom96** [comment]
By the way, thank you for working on this @tmm1. It's a huge help!

**Comment by Varriount** [comment]
I don't mind switching to an externally hosted service, as long as out can support the current platforms and architectures we test (ARM, etc)

**Comment by tmm1** [comment]
> Does Appveyor support IRC notifications?

Doesn't look like it: https://github.com/appveyor/ci/issues/88. You can install arbitrary packages and run any command so it wouldn't be too hard to wire up IRC notifications yourself.

> Does appveyor give us a way to download the resulting binaries/docs?

Yes, this is already hooked up. The built nim.exe is uploaded as an artifact.

> I don't mind switching to an externally hosted service, as long as out can support the current platforms and architectures we test (ARM, etc)

It's possible we could do ARM builds with a cross-compiler on travis or circle, but it might be simpler to continue using the buildbots for that.

**Comment by Varriount** [comment]
Hm. I'd like to avoid a split between build processes.

**Comment by tmm1** [comment]
Only a handful of failures left in here. There's also some sort of issue with `tester.exe --pendantic` which is causing the build not to fail when there are failing tests.

File: tests/testament/tester.nim [hunk 1]
```diff
@@ -104,6 +105,12 @@ proc callCompiler(cmdTemplate, filename, options: string,
   elif suc =~ pegSuccess:
     result.err = reSuccess
 
+  if result.err == reNimcCrash and
+     ("Your platform is not supported" in result.msg or
+      "cannot open 'sdl'" in result.msg or
+      "cannot open 'opengl'" in result.msg):
```
Comment (Araq) [inline tests/testament/tester.nim]: Er, nah. Tests which use SDL or opengl need to move out of the testsuite instead.

Comment (tmm1) [inline tests/testament/tester.nim]: Where should they be moved to?

Comment (dom96) [inline tests/testament/tester.nim]: To the [sdl2](https://github.com/nim-lang/sdl2) and [opengl](https://github.com/nim-lang/opengl) repos I guess.

**Comment by dom96** [comment]
Has this been merged before it was fully functioning?

**Comment by tmm1** [comment]
The builds are functioning as expected, but there are a number of unrelated test failure regressions that need to be resolved in follow-up PRs.

**Pull request merged**